packages/grpc-js/src/subchannel.ts

```diff
@@ -467,11 +467,13 @@ export class Subchannel {
            * if a connection is successfully established through the proxy.
            * If the proxy is not used, these connectionOptions are discarded
            * anyway */
-          connectionOptions.servername = getDefaultAuthority(
+          const targetPath = getDefaultAuthority(
             parseUri(this.options['grpc.http_connect_target'] as string) ?? {
               path: 'localhost',
             }
           );
+          const hostPort = splitHostPort(targetPath);
+          connectionOptions.servername = (hostPort !== null) ? hostPort.host : targetPath;
```

Comment:
This can be simpler: `connectionOptions.servername = hostPort?.host ?? targetPath`. The same change can be made in the http_proxy changes.

Reply:
Thanks for review.